h2/src/main/org/h2/result/LocalResultFactory.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2004-2018 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (http://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.result;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import org.h2.engine.Session;
+import org.h2.expression.Expression;
+import org.h2.message.DbException;
+import org.h2.value.DataType;
+import org.h2.value.Value;
+
+/**
+ * Creates local result.
+ */
+public abstract class LocalResultFactory {
+    /**
+     * Default implementation of local result factory.
+     */
+    public static final LocalResultFactory DEFAULT = new DefaultLocalResultFactory();
+
+    /**
+     * Create a local result object.
+     *
+     * @param session the session
+     * @param expressions the expression array
+     * @param visibleColumnCount the number of visible columns
+     * @return object to collect local result.
+     */
+    public abstract LocalResult create(Session session, Expression[] expressions, int visibleColumnCount);
+
+    /**
+     * Create a local result object.
+     * @return object to collect local result.
+     */
+    public abstract LocalResult create();
+
+    /**
+     * Default implementation of local result factory.
+     */
+    static final class DefaultLocalResultFactory extends LocalResultFactory {
```

Comment:
I think that this class should be `private` (but with empty constructor with default visibility).

Reply:
Fixed